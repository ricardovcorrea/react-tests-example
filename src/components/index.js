export * from './simpleButton';
export * from './simpleHeader';
