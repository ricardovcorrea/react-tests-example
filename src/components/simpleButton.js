export const SimpleButton = (props) => {
  return (
    <button className={'simple-button'} onClick={props.onClick}>
      {props.label}
    </button>
  );
};
