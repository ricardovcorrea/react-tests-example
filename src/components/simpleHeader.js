export const SimpleHeader = (props) => {
    return (
      <div className={"simple-header"}>
        <h1>{props.labels.title}</h1>
  
        <span id={'change-language-button'} onClick={props.onClickChangeLanguage}>
          {props.labels.changeLanguage}
        </span>
      </div>
    );
  };
  