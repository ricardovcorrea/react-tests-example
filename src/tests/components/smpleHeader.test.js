import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import { SimpleHeader } from '../../components/simpleHeader';

describe('SimpleHeader', () => {
  let sut;
  let props;

  beforeEach(() => {
    props = {
      labels: {},
    };
  });

  afterEach(() => {
    if (typeof sut?.unmount === 'function') {
      sut.unmount();
    }
  });

  it('should be a function', () => {
    expect(typeof SimpleHeader).toBe('function');
  });

  it('should render correct html', () => {
    props.labels.title = 'mock title';
    props.labels.changeLanguage = 'mock change language';

    const { container } = render(<SimpleHeader {...props} />);

    expect(container).toMatchSnapshot();
  });

  describe('if the change language span is clicked', () => {
    it('should call onClick callback 1 time', () => {
      props.onClickChangeLanguage = jest.fn();

      render(<SimpleHeader {...props} />);

      const span = document.getElementById('change-language-button');

      fireEvent.click(span);

      expect(props.onClickChangeLanguage).toHaveBeenCalledTimes(1);
    });
  });
});
