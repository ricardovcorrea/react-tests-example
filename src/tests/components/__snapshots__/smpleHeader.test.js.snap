// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleHeader should render correct html 1`] = `
<div>
  <div
    class="simple-header"
  >
    <h1>
      mock title
    </h1>
    <span
      id="change-language-button"
    >
      mock change language
    </span>
  </div>
</div>
`;
